### PR TITLE
[dtest] Fix API incompatibilities in docker harness

### PR DESCRIPTION
**Pull request by vpranckaitis**
**What this PR does / why we need it**:

Fixes a few API incompatibilities in dtest docker harness

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

**Comment by vpranckaitis** [comment]
@rallen090 Adding you for review since it seems that you were the author of the commit in which the endpoints were removed https://github.com/m3db/m3/commit/d86e3ef2bb3419ac45094257b7e9dc7aeffaf004#diff-bd7fae5ad87fc673eb1b8893132d36c40c8c0efb0cec620cd366d8073a5c8162L133-L150

File: src/cmd/tools/dtest/docker/harness/resources/coordinator.go [hunk 1]
```diff
@@ -330,6 +320,26 @@ func (c *coordinator) WriteCarbon(
 	// return nil
 }
 
+func makePostRequest(logger *zap.Logger, url string, body proto.Message) (*http.Response, error) {
+	data := bytes.NewBuffer(nil)
+	if err := (&jsonpb.Marshaler{}).Marshal(data, body); err != nil {
+		logger.Error("failed to marshal", zap.Error(err))
+
+		return &http.Response{}, fmt.Errorf("failed to marshal: %w", err)
```
Comment (rallen090) [inline src/cmd/tools/dtest/docker/harness/resources/coordinator.go:328]: Maybe just return `nil` in these error cases?

**Review by rallen090 — APPROVED** [review]
LGTM

File: src/cmd/tools/dtest/docker/harness/resources/coordinator.go (hunk 1 above)
Comment (vpranckaitis) [reply src/cmd/tools/dtest/docker/harness/resources/coordinator.go]: https://github.com/m3db/m3/pull/2849/commits/f537edccd8d8b4efa432e197d58f7a8d3f98a3c0

**Pull request merged**